### PR TITLE
Шаблонизируй это

### PR DESCRIPTION
**Pull request by elenagradovich**
---
:mortar_board: [Шаблонизируй это](https://up.htmlacademy.ru/ecmascript/14/user/425039/tasks/4)

:boom: https://htmlacademy-ecmascript.github.io/425039-big-trip-14/4/

### CODE REVIEW

File: src/mock/waypoint.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+const generateType =  () => {
+  return types[getRandomValue(types.length - 1)];
+};
+
+export const generateWaypoint = () => {
+  const type = generateType();
+  const offers =  getOffers(type);
+  return {
+    base_price: getRandomValue(100),
```
Comment (KirillKolomin) [inline src/mock/waypoint.js:29]: Why do you use camel_case here, while the last app's written with camelCase?

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { offerList } from'./const';
+
+export const getRandomValue = (maxLength = 1, minLength = 0) => {
+  const lower = Math.ceil(Math.min(minLength, maxLength));
+  const upper = Math.floor(Math.max(minLength, maxLength));
+
+  return Math.floor(lower + Math.random() * (upper - lower + 1));
+};
+
+export const favoriteClass = (is_favorite) => {
```
Comment (KirillKolomin) [inline src/utils.js:11]: why this func is in utils? utils' methods can be used in every part of this app. But the `favoriteClass` only in one view.

Comment (KirillKolomin) [reply src/utils.js:11]: also name methods as verbs

File: src/views/trip-edit-event.js [hunk 3]
```diff
@@ -1,75 +1,85 @@
...
+import dayjs from 'dayjs';
+import { getDate, getOffers } from '../utils';
+import { typeIcons, types } from '../const';
+
+export const tripEditEventTemplate = (event = {}) => {
+  const mockType = 'flight';
+  const mockOffers = getOffers(mockType);
+
+  const {
+    base_price = '30',
```
Comment (KirillKolomin) [inline src/views/trip-edit-event.js:10]: you should define it via mocks

Comment (KirillKolomin) [reply src/views/trip-edit-event.js:10]: once you connect with the server side you must to rewrite all files where there is such kinda things

File: src/views/trip-edit-event.js [hunk 4]
```diff
@@ -1,75 +1,85 @@
-export const tripEditEventTemplate = () => {
-  return `<form class="event event--edit" action="#" method="post">
+import dayjs from 'dayjs';
+import { getDate, getOffers } from '../utils';
+import { typeIcons, types } from '../const';
+
+export const tripEditEventTemplate = (event = {}) => {
+  const mockType = 'flight';
```
Comment (KirillKolomin) [inline src/views/trip-edit-event.js:6]: all mocks to  src/mocks

File: src/views/trip-edit-event.js [hunk 5]
```diff
@@ -1,75 +1,85 @@
...
+                    <span class="event__offer-${offerType}">${offer.price}</span>
+                  </label>
+              </div>`).join(' ');
+  };
+  const createPhotoTemplate = (urlList) => {
+    return urlList.map((url) => `<img class="event__photo" src="${url}" alt="Event photo">`).join(' ');
+  };
+
+  const eventPrice = offers.reduce(((sum, currentValue) =>
+    +sum + +currentValue.price), base_price);
```
Comment (KirillKolomin) [inline src/views/trip-edit-event.js:48]: why do you coerce it to number `+sum` `+currentValue.price`? Do you mean that they can be not numbers?

Comment (elenagradovich) [reply src/views/trip-edit-event.js]: sum, currentValue.price are numbers, but the result is NaN, if i don't convert them obviously. I decided to delete this 'eventPrice' and count price later, when I would choose offers and call handlers in future

File: src/views/trip-edit-event.js [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
     type = mockType,
     offers = mockOffers,
-    destination = {
-      description: `Chamonix-Mont-Blanc (usually shortened to Chamonix) is a resort area near the junction of France,
-                    Switzerland and Italy. At the base of Mont Blanc, the highest summit in the Alps,
-                    it's renowned for its skiing.`,
-      name: 'Chamonix',
-      pictures: ['http://picsum.photos/248/152?r=23', 'http://picsum.photos/248/152?r=45'],
-    },
   } = event;
```
Comment (KirillKolomin) [inline src/views/trip-edit-event.js:22]: does it work? you do destructurization twice. =generateWaypoint() and =event

File: src/views/trip-edit-event.js [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
-  const mockType = 'flight';
-  const mockOffers = getOffers(mockType);
+  const {
+    basePrice: mockBasePrice,
+    dateFrom: mockDateFrom,
+    dateTo: mockDateTo,
+    destination: mockDestination,
+    type: mockType,
+    offers: mockOffers,
+  } = generateWaypoint();
```
Comment (KirillKolomin) [inline src/views/trip-edit-event.js:13]: Please avoid such kinda mock data generation from non mock files

File: src/views/trip-event.js [hunk 8]
```diff
@@ -1,31 +1,49 @@
-export const tripEventTemplate = () => {
+import { getDate, getPeriod } from '../utils';
+import { typeIcons } from '../const';
+
+export const tripEventTemplate = (props) => {
```
Comment (KirillKolomin) [inline src/views/trip-event.js:4]: Please name variables more clear, it's very important. The `props` is equal to `a`, `b`, `test`, `asdasd` etc., variables.

Comment (elenagradovich) [reply src/views/trip-event.js]: renamed

**Pull request merged**